### ISSUE TITLE
%_enable_debug_packages can cause debuginfo on noarch packages

### ISSUE THREAD

**Issue by pmatilai** [rpm-software-management/rpm#3115]
Commit 8535694599ee7f35747d44e2ea0a62dc5e8880e5 hinged the entire debuginfo generation logic around %_enable_debug_packages, including getting it right (disabled) for noarch packages from the platform configuration macros. The problem with that, %_enable_debug_packages is *intended* to be a distro/user configuration knob, whose configuration is read after the platform config. So when a distro explicitly enables sets `%_enable_debug_packages 1` it overrides it even for noarch, causing build failures. 

Luckily its easy to work around: add back the ifnarch test to %debug_macro (or comment out %_enable_debug_packages from distro config for now). But this clearly shows that setting this in platform configuration doesn't cut it, we need to rethink our approach. %_enable_debug_packages is this high-level knob where 0 means "never" and 1 means "create when possible/relevant", and we currently interpret as "always" instead.

**Comment by berolinux** [comment]
Even with this patch, it can still be triggered by using passing `--target x86_64` while building a package that switches to noarch with `BuildArch: noarch`. (`mock` does this, for example)
Probably the relevant bits of the macros are now parsed before `BuildArch` but after `--target`.

**Comment by pmatilai** [comment]
The ticket is still open for a reason, the linked PR was just band-aid as it states.
You mean 'rpmbuild -bb --target x86_64 noarch.spec' ? Why would you do *that*?

**Comment by pmatilai** [comment]
I mean, I know mock always passes --target, and that's been the right thing to do (and still isn't wrong, it's just no longer strictly needed). But it passes --target noarch when building noarch packages, and --target <host> normally.

**Comment by berolinux** [comment]
I agree that `rpmbuild -bb --target x86_64 noarch.spec` is nonsense, but (at least here, and I don't see anything too unusual in the config) mock does that.
Probably best to just not make mock pass --target unless it's crosscompiling.

**Comment by pmatilai** [comment]
Can you point me to a case where it does that?

**Comment by hroncok** [comment]
https://github.com/rpm-software-management/mock/blob/3560d386d32e0d96e50f1495a0ee66c0e9d3fe55/mock/py/mockbuild/backend.py#L744

https://github.com/rpm-software-management/mock/blob/3560d386d32e0d96e50f1495a0ee66c0e9d3fe55/mock/py/mockbuild/backend.py#L43

https://github.com/rpm-software-management/mock/blob/3560d386d32e0d96e50f1495a0ee66c0e9d3fe55/mock/py/mockbuild/config.py#L264

**Comment by pmatilai** [comment]
Oh, thanks but I actually meant a reproducer use-case or such but okay I can reproduce now.
It's pretty weird because under koji, the --target is noarch in mock as it should (eg https://kojipkgs.fedoraproject.org//packages/python-pluggy/1.3.0/3.fc40/data/logs/noarch/build.log) but locally it sets x86_64.

**Comment by berolinux** [comment]
It happens here if I `mock --rebuild any.src.rpm`, nothing special required. Probably koji detects noarch packages and then puts forcearch into the config or something.
